### PR TITLE
BTAT-5712 [JR] - Add predicate for opt-out journey

### PR DESCRIPTION
**Pull request by Jxr227**

### CODE REVIEW

File: app/controllers/predicates/OptOutPredicate.scala [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    implicit val req: User[A] = request
+
+    val getSessionAttribute: String => Option[String] = req.session.get
+
+    (getSessionAttribute(businessName), getSessionAttribute(inflightMandationStatus), getSessionAttribute(mandationStatus)) match {
+      case (_, _, Some(NonMTDfB.value)) =>
+        //TODO error page provided by BTAT-5714
+        Future.successful(Left(errorHandler.showInternalServerError))
+      case (_, Some("true"), _) =>
+        Future.successful(Left(errorHandler.showInternalServerError))
```
Comment (HarryGull) [inline app/controllers/predicates/OptOutPredicate.scala]: Are we sure that where the mandation status is inflight (pending) we don't want to display a graceful error as we seemed to do this on other services? E.g. "You cannot opt out now as you have already requested to opt out which is pending"?

**Comment by Jxr227** [comment]
https://github.com/hmrc/app-config-base/pull/1991 
Should be merged along with this to ensure the config is correct

File: app/controllers/predicates/OptOutPredicate.scala [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+import play.api.mvc.Results.Redirect
+import play.api.mvc.{ActionRefiner, Result}
+import services.VatSubscriptionService
+import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.HeaderCarrierConverter
+
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class OptOutPredicate @Inject()(vatSubscriptionService: VatSubscriptionService,
```
[the remark this answers is not in the document]
Comment (Jxr227) [reply app/controllers/predicates/OptOutPredicate.scala]: Fixed in latest push

File: app/controllers/predicates/OptOutPredicate.scala (hunk 1 above)
Comment (Jxr227) [reply app/controllers/predicates/OptOutPredicate.scala]: Discussed and for now we have to exception, possible improvement in future

**Pull request merged**